doctor/parsers.py

```diff
@@ -243,15 +243,21 @@ def parse_form_and_query_params(req_params, sig_params):
                         custom_parser))
                 custom_parser = None
 
```

Comment:
Moved this into the else block below as we don't need these values if a custom_parser is defined.